integration/lsp.test.ts

```diff
@@ -0,0 +1,125 @@
+function make_request(data) {
+    data.jsonrpc = "2.0";
+    data.id = 1;
+    const json = JSON.stringify(data);
+    return `Content-Length: ${json.length}\r\n\r\n${json}`;
+}
+
+function parse_response(data) {
+    let json = JSON.parse(data.split('\r\n')[2]);
+    delete json.id;
+    delete json.jsonrpc;
+    return json;
+}
+
+test('initialize', async () => {
+    import('@influxdata/flux-lsp-node')
+    .then(async ({Server}) => {
+        const expected = {
+            "result": {
+                "capabilities": {
+                    "textDocumentSync":1,
+                    "referencesProvider":true,
+                    "definitionProvider":true,
+                    "renameProvider":true,
+                    "foldingRangeProvider":true,
+                    "documentSymbolProvider":true,
+                    "documentFormattingProvider":true,
+                    "completionProvider": {
+                        "resolveProvider":true,
+                        "triggerCharacters": [".",":","(",",","\""]
+                    },
+                    "signatureHelpProvider": {
+                        "triggerCharacters": ["("],
+                        "retriggerCharacters": ["("]
+                    },
+                    "hoverProvider":true
+                }
+            }
+        };
+
+        const server = new Server(false, false);
+        const request = make_request({method: "initialize"});
+
+        const response = await server.process(request);
+        const error = response.get_error();
+        expect(error).toBe(undefined);
+        const message = parse_response(response.get_message());
+        expect(message).toStrictEqual(expected);
```

Comment:
For readability, it's generally nice to keep the `expect`s together, IMO.

Reply:
Yeah, in this case, I want to check the error first, and if there is no error, continue on with the message (the thing I actually care about). This makes it easier to debug when a panic happens inside the wasm (an issue I had initially). I'm happy to straight up remove these lines as well, because they aren't needed now anyway. Thoughts?

Reply:
Would there ever be a case where **both** `error` and `message` would be defined? If not, then you can get rid of the error check in most places since a successful `message` check would imply an empty error. I do, however think it would be worth checking it explicitly __somewhere__, so either leave it in one of these tests or have explicit test cases that check when the error should and should not be there. Right?